08-compiler-features.tex

```diff
@@ -142,81 +142,347 @@ \section{Dead Code Elimination}
+<type>
+delimiter : String -&gt; Array&lt;String&gt;
+</type>
+\end{lstlisting}
+
+IDEs can parse this to recognize that the called function requires one argument named \ic{delimiter} of type \type{String} and returns an \type{Array<String>}.
+
+\trivia{Problems with the output structure}{We acknowledge that the current format requires a bit of manual parsing which can be annoying. In the future we might look into providing a more structured output, especially for functions.}
+
+\subsection{Type path completion}
+\label{cr-completion-type-path}
 
-This parameter determines the file in which we want some completion and the byte position (not char positions) of the cursor in the file. In the example given, you should get 73 if you count characters with Windows' CRLF lines ending.
+Type path completion can occur in \tref{import declarations}{type-system-import}, \tref{using declarations}{lf-static-extension} or any place a type is referenced. We can identify three different cases:
 
+\paragraph{package completion}
 
-Haxe will perform all the parsing and typing as it would do for normal compilation, except that it will not generate any code. If it reaches the byte 73 in the file \ic{Test.hx} during compilation, it will output the information about the type of the expression on the left of the dot.
+This lists all sub-packages of the haxe package, as well as all modules in that package:
 
+\begin{lstlisting}
+import haxe.|
+\end{lstlisting}
+
+\begin{lstlisting}
+<list>
+<i n="CallStack"><t></t><d></d></i>
+<i n="Constraints"><t></t><d></d></i>
+<i n="DynamicAccess"><t></t><d></d></i>
+<i n="EnumFlags"><t></t><d></d></i>
+<i n="EnumTools"><t></t><d></d></i>
+<i n="Http"><t></t><d></d></i>
+<i n="Int32"><t></t><d></d></i>
+<i n="Int64"><t></t><d></d></i>
+<i n="Json"><t></t><d></d></i>
+<i n="Log"><t></t><d></d></i>
+<i n="PosInfos"><t></t><d></d></i>
+<i n="Resource"><t></t><d></d></i>
+<i n="Serializer"><t></t><d></d></i>
+<i n="Template"><t></t><d></d></i>
+<i n="Timer"><t></t><d></d></i>
+<i n="Ucs2"><t></t><d></d></i>
+<i n="Unserializer"><t></t><d></d></i>
+<i n="Utf8"><t></t><d></d></i>
+<i n="crypto"><t></t><d></d></i>
+<i n="ds"><t></t><d></d></i>
+<i n="extern"><t></t><d></d></i>
+<i n="format"><t></t><d></d></i>
+<i n="io"><t></t><d></d></i>
+<i n="macro"><t></t><d></d></i>
+<i n="remoting"><t></t><d></d></i>
+<i n="rtti"><t></t><d></d></i>
+<i n="unit"><t></t><d></d></i>
+<i n="web"><t></t><d></d></i>
+<i n="xml"><t></t><d></d></i>
+<i n="zip"><t></t><d></d></i>
+</list>
+\end{lstlisting}
 
-In our case, that would be \type{String}. The exception being that in the case of an anonymous object or a class instance, the compiler will, instead, list the different fields in an XML and print them to a possible output:
+
+\paragraph{import module completion}
+
+This lists all \tref{sub-types}{type-system-module-sub-types} of the module \type{haxe.Unserializer} as well as all its public static fields (because these can be imported as well):
```

Comment:
consider replacing "as well as" with "and" since you also use "as well" in the parthesis